eng/common/pipelines/templates/jobs/archetype-spec-gen-sdk.yml

```diff
@@ -0,0 +1,199 @@
+parameters:
+  - name: SpecRepoUrl
+    type: string
+  - name: SdkRepoUrl
+    type: string
+  - name: SpecRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: SdkRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: ConfigType
+    type: string
+    values:
+      - 'TypeSpec'
+      - 'OpenAPI'
+    default: 'TypeSpec'
+    displayName: 'API specification type'
+  - name: ConfigPath
+    type: string
+    default: 'specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml'
+    displayName: 'Path to API specification file (TypeSpec/OpenAPI)'
+  - name: SkipPullRequestCreation
+    type: boolean
+    default: false
+
+jobs:
+- job:
+  displayName: 'SDK Generation'
+
+  variables:
+    - template: /eng/pipelines/templates/variables/image.yml
+    - name: NodeVersion
+      value: '22.13.x'
+    - name: PythonVersion
+      value: '3.13'
+    - name: SpecGenSdkVersion
+      value: 'latest'
+    - name: SdkArtifactName
+      value: SDK_Artifact
+
+  pool:
+    name: $(LINUXPOOL)
+    vmImage: $(LINUXVMIMAGE)
+
+  steps:
+    - checkout: none
+
+    - pwsh: |
+        $urlPattern = '^https://github\.com/(?<organization>[^/]+)/(?<repository>[^/]+)'
+        if ('${{ parameters.SpecRepoUrl }}' -match $urlPattern) {
+          $specRepoOwner = $Matches['organization']
+          Write-Host "##vso[task.setvariable variable=SpecRepoOwner]$specRepoOwner"
+          Write-Host "SpecRepoOwner variable set to: $specRepoOwner"
+
+          $specRepoName = $Matches['repository']
+          Write-Host "##vso[task.setvariable variable=SpecRepoName]$specRepoName"
+          Write-Host "SpecRepoName variable set to: $specRepoName"
+
+          $specRepoDirectory = "$(System.DefaultWorkingDirectory)/$specRepoName"
+          Write-Host "##vso[task.setvariable variable=SpecRepoDirectory]$specRepoDirectory"
+          Write-Host "SpecRepoDirectory variable set to: $specRepoDirectory"
+        }
+
+        if ('${{ parameters.SdkRepoUrl }}' -match $urlPattern) {
+          if ('${{ parameters.SpecRepoUrl }}'.EndsWith('-pr') -and (-not '${{ parameters.SdkRepoUrl }}'.EndsWith('-pr'))) {
+            Write-Host "SdkRepoUrl must be a private repository if SpecRepoUrl is a private repository."
+            Exit 1
+          }
+
+          $sdkRepoOwner = if ('$(SdkRepoOwner)' -eq '') { $Matches['organization'] } else { '$(SdkRepoOwner)' }
```

Comment:
I suspect you will need to remove this if condition if it is not set to something before this step, otherwise DevOps will not replace it with `''` and you will end up with a bogus owner.

Reply:
`SdkRepoOwner` is set as a pipeline variable set to empty string, but also setable at queue time so users can change it. So yes it will be set to empty initially.

Reply:
Why does this need to be a pipeline variable? I understand we have SdkRepoUrl but I don't know why we would ever set this in the pipeline. We should probably clean that up unless we have a concrete scenario where it is interesting.

Reply:
I want to minimize the number of variables we have to setup on the pipeline UI side itself and this one doesn't seem like it would be needed in most cases. If we end up wanting to support forks we should promote to a parameter. I however wouldn't do that for now.